docs/lib/datastore/fragments/ios/real-time/observe-snippet.md

```diff
@@ -1,17 +1,26 @@
 ```swift
-let postSubscription = Amplify.DataStore
-    .publisher(for: Post.self)
-    .sink(receiveCompletion: { completion in
-        if case .failure(let error) = completion {
-            print("Subscription received error - \(error.localizedDescription)")
-        }
-    }) { changes in
-    // handle incoming changes
-    print("Subscription received mutation: \(changes)")
+// In your type declaration, declare a cancellable to hold onto the subscription
+var postsSubscription: AnyCancellable?
+var postsPublisher: AnyPublisher<MutationEvent, DataStoreError>?
+
+// Then in the body of your code, subscribe to the publisher
+func subscribeToPosts() {
+    postsPublisher = Amplify.DataStore.publisher(for: Todo.self)
+    postsSubscription = postsPublisher?
```

Comment:
Since we are not using the `publisher`, is there any reason to create a variable for it?

Reply:
D'oh! Meant to remove that. Fixed.